components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/soaptorest/template/RESTToSOAPMsgTemplate.java

```diff
@@ -0,0 +1,133 @@
+import org.json.simple.JSONArray;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.APIManagerConfigurationService;
+import org.wso2.carbon.apimgt.impl.internal.ServiceReferenceHolder;
+import org.wso2.carbon.apimgt.impl.template.ConfigContext;
+
+import java.io.File;
+import java.io.StringWriter;
+import java.util.Map;
+
+/**
+ *  velocity template to generate in sequences and out sequences
+ */
+public class RESTToSOAPMsgTemplate {
+
+    private static final Log log = LogFactory.getLog(RESTToSOAPMsgTemplate.class);
+
+    private String velocityLogPath = null;
+    private static final String IN_SEQ_TEMPLATE_FILE = "soap_to_rest_in_seq_template";
+    private static final String OUT_SEQ_TEMPLATE_FILE = "soap_to_rest_out_seq_template";
+
+    /**
+     * gets in sequence for the soap operation
+     *
+     * @param mapping soap to rest mapping json
+     * @param method http method for the resource
+     * @param soapAction soap action for the operation
+     * @param namespace soap namespace for the sequence
+     * @return in sequence string
+     */
+    public String getMappingInSequence( Map<String, String> mapping, String method, String soapAction, String namespace, JSONArray array) {
+
+        ConfigContext configcontext = new SOAPToRESTConfigContext(mapping, method, soapAction, namespace, array);
+        StringWriter writer = new StringWriter();
+        try {
+            VelocityContext context = configcontext.getContext();
+            context.internalGetKeys();
+
+            VelocityEngine velocityengine = new VelocityEngine();
+            if (!"not-defined".equalsIgnoreCase(getVelocityLogger())) {
+                velocityengine.setProperty( RuntimeConstants.RUNTIME_LOG_LOGSYSTEM_CLASS,
+                        "org.apache.velocity.runtime.log.Log4JLogChute" );
+                velocityengine.setProperty( "runtime.log.logsystem.log4j.logger", getVelocityLogger());
+            }
+
+            velocityengine.init();
+            Template t = velocityengine.getTemplate(this.getInSeqTemplatePath());
+
+            t.merge(context, writer);
+        } catch (Exception e) {
+            log.error("Velocity Error", e);
+        }
+        return writer.toString();
+    }
+
+    /**
+     * gets out sequence for the soap operation
+     *
+     * @return out sequence for the converted soap operation
+     */
+    public String getMappingOutSequence() {
+
+        ConfigContext configcontext = new SOAPToRESTConfigContext();
+        StringWriter writer = new StringWriter();
+        try {
+            VelocityContext context = configcontext.getContext();
+            context.internalGetKeys();
+
+            VelocityEngine velocityengine = new VelocityEngine();
+            if (!"not-defined".equalsIgnoreCase(getVelocityLogger())) {
+                velocityengine.setProperty( RuntimeConstants.RUNTIME_LOG_LOGSYSTEM_CLASS,
+                        "org.apache.velocity.runtime.log.Log4JLogChute" );
```

Comment:
Use a String constant here for org.apache.velocity.runtime.log.Log4JLogChute

Reply:
fixed in e18f43f5dee9ed43a3239af62f38d67c72c78fa2